File: app/src/main/java/com/example/myapplication/LoginActivity.java
package com.example.myapplication;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import androidx.work.Constraints;
import androidx.work.Data;
import androidx.work.NetworkType;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkManager;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import com.example.myapplication.databinding.ActivityLoginBinding;


public class LoginActivity extends AppCompatActivity {
    private ActivityLoginBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityLoginBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());









        Button login = binding.loginButton;
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                EditText username = (EditText) findViewById(R.id.editTextTextPersonName);
                EditText password = (EditText) findViewById(R.id.editTextTextPassword);

                Constraints.Builder builder = new Constraints.Builder()
                        .setRequiredNetworkType(NetworkType.CONNECTED);
                Data.Builder data = new Data.Builder();
                data.putString("username", username.getText().toString());
                data.putString("password", password.getText().toString());
                OneTimeWorkRequest testWorkRequest =
                        new OneTimeWorkRequest.Builder(LoginWorker.class)
                                .addTag("Login")
                                .setInputData(data.build())
                                .setConstraints(builder.build())
                                .build();
                WorkManager.getInstance(getApplicationContext()).enqueue(testWorkRequest);

                /*
                while(true) {
                    WorkManager instance = WorkManager.getInstance(getApplicationContext());
                    ListenableFuture<List<WorkInfo>> status = instance.getWorkInfosByTag("Login");
                    List<WorkInfo> workInfoList = null;

                    try {
                        workInfoList = status.get();
                    }
                    catch (Exception e) {
                        Log.d("Tag","Not Found");
                    }

                    WorkInfo first=workInfoList.get(0);
                    if(first.toString().equalsIgnoreCase("SUCCEEDED")) {
                        Log.d("Login Result","Succeeded");
                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                        startActivity(intent);
                        break;
                    }
                    else if(first.toString().equalsIgnoreCase("FAILED")) {
                        Log.d("Login Result","Failed");
                        break;
                    }
                    else if(first.toString().equalsIgnoreCase("RUNNING")) {
                        Log.d("Login Result","Running");
                    }

                }
                */




            }
        });
        Button register = binding.registerButton;
        register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                EditText username = (EditText) findViewById(R.id.editTextTextPersonName);
                EditText password = (EditText) findViewById(R.id.editTextTextPassword);

                Constraints.Builder builder = new Constraints.Builder()
                        .setRequiredNetworkType(NetworkType.CONNECTED);
                Data.Builder data = new Data.Builder();
                data.putString("username", username.getText().toString());
                data.putString("password", password.getText().toString());
                OneTimeWorkRequest regWorkRequest =
                        new OneTimeWorkRequest.Builder(RegisterWorker.class)
                                .addTag("Register")
                                .setInputData(data.build())
                                .setConstraints(builder.build())
                                .build();
                WorkManager.getInstance(getApplicationContext()).enqueue(regWorkRequest);
            }});}}

                /*
                while(true) {
                    WorkManager instance = WorkManager.getInstance(getApplicationContext());
                    ListenableFuture<List<WorkInfo>> status = instance.getWorkInfosByTag("Register");
                    List<WorkInfo> workInfoList = null;

                    try {
                        workInfoList = status.get();
                    }
                    catch (Exception e) {
                        Log.d("Tag","Not Found");
                    }

                    WorkInfo first=workInfoList.get(0);
                    if(first.toString().equalsIgnoreCase("SUCCEEDED")) {
                        Log.d("Registration Result","Succeeded");
                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                        startActivity(intent);
                        break;
                    }
                    else if(first.toString().equalsIgnoreCase("FAILED")) {
                        Log.d("Registration Result","Failed");
                        break;
                    }
                    else if(first.toString().equalsIgnoreCase("RUNNING")) {
                        Log.d("Registration Result","Running");
                    }
                    try {
                        Thread.sleep(3000);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }

                }
            }



                while(true){
                if(workInfo.State==WorkInfo.State.SUCCEEDED){

                }
                else if(WorkInfo.State==WorkInfo.State.FAILED){
                    Log.d("Registration", "Failed");
                });
            */

File: app/src/main/java/com/example/myapplication/LoginActivity.kt
package com.example.myapplication
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.work.*
import com.example.myapplication.databinding.ActivityLoginBinding
import com.google.android.material.snackbar.Snackbar

class LoginActivity : AppCompatActivity() {
    private var binding: ActivityLoginBinding? = null
    private val workManager=WorkManager.getInstance(application)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        val login = binding!!.loginButton
        login.setOnClickListener {
            val username = findViewById<View>(R.id.editTextTextPersonName) as EditText
            val password = findViewById<View>(R.id.editTextTextPassword) as EditText
            val builder = Constraints.Builder()
                    .setRequiredNetworkType(NetworkType.CONNECTED)
            val data = Data.Builder()
            data.putString("username", username.text.toString())
            data.putString("password", password.text.toString())
            val testWorkRequest = OneTimeWorkRequest.Builder(LoginWorker::class.java)
                    .addTag("Login")
                    .setInputData(data.build())
                    .setConstraints(builder.build())
                    .build()
            workManager.enqueue(testWorkRequest)
            workManager.getWorkInfosByTagLiveData("Login").observe(this){ workInfos ->
                //Log.d("Workstatus",workInfo[0].state.toString());
                val workInfo=workInfos[0]
                if(workInfo?.state == WorkInfo.State.SUCCEEDED) {
                    Log.d("Login Result","Succeeded");
                    val intent = Intent(this, MainActivity::class.java);
                    startActivity(intent);

                }
                else if(workInfo?.state==WorkInfo.State.FAILED){
                    var failmsg="Bad Login";
                    val mySnackbar = Snackbar.make(findViewById(R.id.editTextTextPassword), failmsg, 3)
                    mySnackbar.show()
                }
            }
        }
        val register = binding!!.registerButton
        register.setOnClickListener {
            val username = findViewById<View>(R.id.editTextTextPersonName) as EditText
            val password = findViewById<View>(R.id.editTextTextPassword) as EditText
            val builder = Constraints.Builder()
                    .setRequiredNetworkType(NetworkType.CONNECTED)
            val data = Data.Builder()
            data.putString("username", username.text.toString())
            data.putString("password", password.text.toString())
            val regWorkRequest = OneTimeWorkRequest.Builder(RegisterWorker::class.java)
                    .addTag("Register")
                    .setInputData(data.build())
                    .setConstraints(builder.build())
                    .build()
            WorkManager.getInstance(applicationContext).enqueue(regWorkRequest)
            Thread.sleep(1_000)
            Log.d("Status of Registration",(WorkManager.getInstance(applicationContext).getWorkInfosByTag("Login").toString()));

        }
    }
}
